datumaro/cli/commands/download.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import argparse
+import logging as log
+import os
+import os.path as osp
+
+from datumaro.components.extractor_tfds import (
+    AVAILABLE_TFDS_DATASETS, TFDS_EXTRACTOR_AVAILABLE, make_tfds_extractor,
+)
+from datumaro.components.project import Environment
+from datumaro.util.os_util import make_file_name
+
+from ..util import MultilineFormatter
+from ..util.errors import CliException
+from ..util.project import generate_next_file_name
+
+
+def build_parser(parser_ctor=argparse.ArgumentParser):
+    builtin_writers = sorted(Environment().converters)
+    if TFDS_EXTRACTOR_AVAILABLE:
+        available_datasets = ", ".join(
+            f'tfds:{name}' for name in AVAILABLE_TFDS_DATASETS)
+    else:
+        available_datasets = "N/A (TensorFlow and/or TensorFlow Datasets " \
+            "are not installed)"
+
+    parser = parser_ctor(
+        help="Download a publicly available dataset",
+        description="""
+        Downloads a publicly available dataset and saves it in a given format.|n
+        |n
+        Currently, the only source of datasets is the TensorFlow Datasets
+        library. Therefore, to use this command you must install TensorFlow &
+        TFDS, which you can do as follows:|n
+        |n
+        |s|spip install datumaro[tf,tfds]|n
+        |n
+        Supported datasets: {}|n
+        |n
+        For information about the datasets, see the TFDS Catalog:
+        <https://www.tensorflow.org/datasets/catalog/overview>.|n
+        |n
+        Supported output formats: {}|n
+        |n
+        Examples:|n
+        - Download the MNIST dataset, saving it in the ImageNet text format:|n
+        |s|s%(prog)s -i tfds:mnist -f imagenet_txt -- --save-images|n
+        |n
+        - Download the VOC 2012 dataset, saving only the annotations in the COCO
+          format into a specific directory:|n
+        |s|s%(prog)s -i tfds:voc/2012 -f coco -o path/I/like/
+        """.format(available_datasets, ', '.join(builtin_writers)),
+        formatter_class=MultilineFormatter)
+
+    parser.add_argument('-i', '--dataset-id', required=True,
+        help="Which dataset to download")
+    parser.add_argument('-f', '--output-format', required=True,
+        help="Output format")
+    parser.add_argument('-o', '--output-dir', dest='dst_dir',
+        help="Directory to save output (default: a subdir in the current one)")
```

Comment:
I see there is no option for manipulating intermediate cache directory for TFDS data, but it can be pretty useful in practice. Don't you think it will be better to just download the data as is or to provide an option like `cache-dir`?

Reply:
The location can be customized with the `TFDS_DATA_DIR` environment variable, so I didn't think it was important enough to add an option for it.